bundles/org.openhab.binding.plex/src/main/java/org/openhab/binding/plex/internal/handler/PlexApiConnector.java

```diff
@@ -0,0 +1,407 @@
+        token = connProps.getToken();
+        host = connProps.getHost();
+        port = connProps.getPort();
+        wsClient = new WebSocketClient();
+        PlexSocket = new PlexSocket();
+    }
+
+    private String getSchemeWS() {
+        return scheme.equals("http") ? "ws" : "wss";
+    }
+
+    public boolean hasToken() {
+        return !token.isBlank();
+    }
+
+    /**
+     * Base configuration for XStream
+     */
+    private void setupXstream() {
+        XStream.setupDefaultSecurity(xStream);
+        xStream.allowTypesByWildcard(
+                new String[] { User.class.getPackageName() + ".**", MediaContainer.class.getPackageName() + ".**" });
+        xStream.setClassLoader(PlexApiConnector.class.getClassLoader());
+        xStream.ignoreUnknownElements();
+        xStream.processAnnotations(User.class);
+        xStream.processAnnotations(MediaContainer.class);
+    }
+
+    /**
+     * Fetch the XML data and parse it through xStream to get a MediaContainer object
+     *
+     * @return
+     */
+    public @Nullable MediaContainer getSessionData() {
+        try {
+            String url = "http://" + host + ":" + String.valueOf(port) + "/status/sessions" + "?X-Plex-Token=" + token;
+            MediaContainer mediaContainer = doHttpRequest("GET", url, getClientHeaders(), MediaContainer.class);
+            return mediaContainer;
+        } catch (Exception e) {
+            logger.warn("An exception occurred while polling the PLEX Server: '{}'", e.getMessage());
+            return null;
+        }
+    }
+
+    /**
+     * Assemble the URL to include the Token
+     *
+     * @param url The url portion that is returned from the sessions call
+     * @return the completed url that will be usable
+     */
+    public String getURL(String url) {
+        String artURL = scheme + "://" + host + ":" + String.valueOf(port + url + "?X-Plex-Token=" + token);
+        return artURL;
+    }
+
+    /**
+     * This method will get an X-Token from the PLEX servers if one is not provided in the bridge config
+     *
+     * @return
+     */
+    public boolean getToken() {
+        try {
+            String url = SIGNIN_URL;
+            String authString = Base64.getEncoder().encodeToString((username + ":" + password).getBytes());
+            Properties headers = getClientHeaders();
+            headers.put("Authorization", "Basic " + authString);
+            User user = doHttpRequest("POST", url, headers, User.class);
+            if (user != null) {
+                logger.debug("PLEX login successful using username/password");
+                token = user.getAuthenticationToken();
+                return true;
+            } else {
+                logger.warn("Invalid credentials for Plex account, please check config");
+                return false;
+            }
+        } catch (Exception e) {
+            logger.warn("An exception occurred while fetching auth token:'{}'", e.getMessage(), e);
```

Comment:
When an error occured during connecting, you could set the ThingStatus to OFFLINE with the ThingStatusDetail to COMMUNICATION_ERROR or CONFIGURATION_ERROR. The error message you are currently logging here can then be set as the thing status detail message. This will be displayed in the UI, showing the user what's wrong. See `updateStatus()` in the ThingHandler.